apps/st2-history/controller.js

```diff
@@ -253,6 +256,46 @@ angular.module('main')
       }
     };
 
+    $scope.rerun = (function () {
+      var rerun = $scope.$new();
+
+      rerun.metaSpec = {
+        type: 'object',
+        properties: {
+          ref: {
+            type: 'string',
+            name: 'Action'
+          }
+        }
+      };
+
+
+      rerun.open = function () {
+        $scope.$root.state.go('^.rerun', {id: $scope.record.id});
+        rerun.payload = _.clone($scope.payload);
```

Comment:
We're cloning the parameters execution has been ran with. We're doing it so user would understand what to expect. If we don't do that, UI would show either an empty field (and will left user wondering what exactly do he want to change here) or `default` value from action schema (which would be misleading).

So, say we have an execution of `core.local` with parameters `{cmd: 'ls -lah', cwd: '/'}`. The `rerun.payload` would be the same as original payload. Then user would change `cwd` to `/home` and the resulting payload sent to the server would be `{cmd: 'ls -lah', cwd: '/'}`.

There is a problem though, the fact that besides 'current value' and 'previous value' there is also a default one (an empty `cwd` in case of running a new action) and with current implementation of re_run API endpoint, I see no way we can restore the original, empty value during rerun.